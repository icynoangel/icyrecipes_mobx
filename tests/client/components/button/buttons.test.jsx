import React from 'react';
import ReactDOM from 'react-dom';
import Buttons from './../../../../src/client/js/components/button/buttons';
import Button from './../../../../src/client/js/components/button/button';
import Modal from './../../../../src/client/js/components/modal/modal';
import {mount} from 'enzyme';
import enzymeToJson from 'enzyme-to-json';

describe('<Buttons />', function() {
  this.props = {
    ItemsStore: {
      getItems: jest.fn()
    }
  };
  this.getComponent = props => {
    return <Buttons {...this.props} />;
  };

  it('Should render <Buttons />, <Modal /> and <ModalConfirmation />', () => {
    const wrapper = mount(this.getComponent());
    expect(enzymeToJson(wrapper)).toMatchSnapshot();
  });

  it('Should call open method from <Modal />', () => {
    const openSpy = jest.spyOn(Modal.prototype, 'open');
    const wrapper = mount(this.getComponent());
    wrapper
      .find(Button)
      .last()
      .simulate('click');
    expect(openSpy).toHaveBeenCalledTimes(1);
  });

  it('Should call onConfirm from <Buttons /> when confirm is pressed', () => {
    const confirmSpy = jest.spyOn(
      Buttons.wrappedComponent.prototype,
      'onConfirm'
    );
    const wrapper = mount(this.getComponent());
    wrapper
      .find(Button)
      .last()
      .simulate('click');
    wrapper.find('[caption="Confirm"]').simulate('click');
    expect(confirmSpy).toHaveBeenCalledTimes(1);
  });

  it('Should call onDeny from <Buttons /> when deny is pressed', () => {
    const denySpy = jest.spyOn(Buttons.wrappedComponent.prototype, 'onDeny');
    const wrapper = mount(this.getComponent());
    wrapper
      .find(Button)
      .last()
      .simulate('click');
    wrapper.find('[caption="Deny"]').simulate('click');
    expect(denySpy).toHaveBeenCalledTimes(1);
  });
});
