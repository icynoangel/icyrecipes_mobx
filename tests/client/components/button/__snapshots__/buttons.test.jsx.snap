// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Buttons /> Should render <Buttons />, <Modal /> and <ModalConfirmation /> 1`] = `
<inject-Buttons-with-ItemsStore
  ItemsStore={
    Object {
      "getItems": [MockFunction],
    }
  }
>
  <Buttons
    ItemsStore={
      Object {
        "getItems": [MockFunction],
      }
    }
  >
    <div
      className="buttons"
    >
      <Button
        caption="Get Items"
        key="button-request-items"
        onClick={[MockFunction]}
        type="primary"
      >
        <button
          className="button --primary"
          onClick={[MockFunction]}
        >
          Get Items
        </button>
      </Button>
      <Button
        caption="Open Modal"
        key="open-confirmation-modal"
        onClick={[Function]}
        type="primary"
      >
        <button
          className="button --primary"
          onClick={[Function]}
        >
          Open Modal
        </button>
      </Button>
      <Modal
        size="small"
      />
    </div>
  </Buttons>
</inject-Buttons-with-ItemsStore>
`;
